dspace-api/src/main/java/org/dspace/content/RelationshipServiceImpl.java

```diff
@@ -346,6 +357,119 @@ public void delete(Context context, Relationship relationship, boolean copyToLef
+     *
+     * @param context           The relevant DSpace context
+     * @param relationship      The relationship which has been created, updated or deleted
+     * @throws SQLException     If something goes wrong
+     */
+    private void updateItemsInRelationship(Context context, Relationship relationship) throws SQLException {
+        // Since this call is performed after creating, updating or deleting the relationships, the permissions have
+        // already been verified. The following updateItem calls can however call the
+        // ItemService.update() functions which would fail if the user doesn't have permission on both items.
+        // Since we allow this edits to happen under these circumstances, we need to turn off the
+        // authorization system here so that this failure doesn't happen when the items need to be update
+        context.turnOffAuthorisationSystem();
+        try {
+            int max = configurationService.getIntProperty("relationship.update.relateditems.max", 5);
+            List<Item> itemsToUpdate = new LinkedList<>();
+            itemsToUpdate.add(relationship.getLeftItem());
+            itemsToUpdate.add(relationship.getRightItem());
+
+            itemsToUpdate = getRelatedItemsForLeftItem(context, relationship.getLeftItem(),
+                                                       relationship, itemsToUpdate, max);
+            itemsToUpdate = getRelatedItemsForRightItem(context, relationship.getRightItem(),
+                                                        relationship, itemsToUpdate, max);
+
+            for (Item item : itemsToUpdate) {
+                if (!item.isMetadataModified()) {
+                    updateItem(context, item);
+                }
+            }
+
+        } catch (AuthorizeException e) {
+            log.error("Authorization Exception while authorization has been disabled", e);
+        }
+        context.restoreAuthSystemState();
+    }
+
+    private List<Item> getRelatedItemsForRightItem(Context context, Item item, Relationship relationship,
+                                                   List<Item> itemsToUpdate, int max)
+        throws SQLException {
+        if (itemsToUpdate.size() >= max) {
+            return itemsToUpdate;
+        }
+        List<RelationshipType> relationshipTypes = new LinkedList<>();
+        EntityType leftType = relationship.getRelationshipType().getLeftType();
+        String entityTypeStringFromMetadata = relationshipMetadataService.getEntityTypeStringFromMetadata(item);
+        EntityType actualEntityType = entityTypeService.findByEntityType(context, entityTypeStringFromMetadata);
+        boolean isLeft = false;
+        if (StringUtils.equalsIgnoreCase(leftType.getLabel(),
+                                         entityTypeStringFromMetadata)) {
+            relationshipTypes = relationshipTypeService.findByEntityType(context, actualEntityType, false);
+        } else {
+            isLeft = true;
```

Comment:
I'm a little confused by the logic in these methods as there's no inline comments or JavaDocs.  If we already *know* that this `item` is the "RightItem", then why do we need to check if it's actually the *left item* (when would `isLeft` ever be "true")?

I'm assuming I'm misunderstanding this code, but I wanted to point out that it's not clear to me.  I would appreciate more inline comments, as I don't understand the logic in this method